src/pages/products/index.tsx

```diff
@@ -1,59 +1,102 @@
+import { WarningWindow } from 'components/warningWindow'
 
 export const Products = () => {
-  const [product, setProduct] = useState<IProduct>({} as IProduct)
-  const { title } = useParams()
-  const { isAuthenticated } = useContext(UserContext)
-  const [description, setDescription] = useState('')
+  const [value, setValue] = useState('')
+  const [products, setProducts] = useState<IProduct[]>([])
+  const [editIndex, setEditIndex] = useState(-1)
+  const [id, setId] = useState<number | undefined>(undefined)
+  const productsWithImages = products.filter(product => value.length >= 2 ? product.title.includes(value) : product.image)
+  const getDescription = (id: number) => {
+    const findProduct = products.find(data => data.id === id)
+
+    return `You're going to remove "${findProduct?.title}" product. Press "Ok" to confirm.`
+  }
+  const description = useMemo(() => getDescription(id!), [id])
 
   const getData = async () => {
-    setProduct(await findProduct(title))
+    setProducts(await getProducts())
   }
 
   useEffect(() => {
     getData()
   }, [])
 
-  const handleSave = async () => {
-   await updateProduct({...product, description: description})
-   getData()
+  const searchCategory = ({ target: { value } }: TTarget) => {
+    setValue(value)
+  }
+
+  const handleRename = async (product: IProduct, name: string) => {
+    await updateProduct({ ...product, title: name })
+    getData()
+
+    setEditIndex(-1)
+  }
+
+  const onConfirm = async () => {
+    await deleteProduct(id!)
+    setId(undefined)
+    getData()
   }
 
   return (
-    isAuthenticated
-      ? (
-        <div className={styles.productInformation} >
-          <span className={styles.title}>
-            TITLE: <EditableText text={product.title} />
-          </span>
-          <span className={styles.price}>
-            $ <EditableText text={String(product.price)} price />
-          </span>
-          <div className={styles.description}>
-            <EditableText onBlur={(description) => setDescription(description)} text={product.description || 'No description'} multiLine />
-          </div>
-          <button type="button" className={styles.save} onClick={handleSave}>SAVE</button>
-        </div>
-      )
-      : (
-        <div className={styles.productInformation} >
-          <span className={styles.title}>
-            TITLE: {product.title}
-          </span>
-          <span className={styles.price}>
-            $ {product.price}
-          </span>
-          <div className={styles.description}>
-            {product.description || 'No description'}
-          </div>
-        </div >
-      )
+    <>
+      <h1 className={styles.heading}>
+        Products
+      </h1>
+      {
+        productsWithImages.length
+        ? <input
+          className={styles.search}
+          type="text"
+          placeholder="Enter at list 2 chars"
+          value={value}
+          onChange={searchCategory}
+        />
+        : ''
```

Comment:
why do you need an empty string?

Reply:
Instead of an empty string, I'll make an empty span. Since if you do productsWithImages.length && input, then when the array of products is empty, in addition to the text No products, it also shows 0.

Reply:
Use ternary only when you need `else` branch otherwise use `&&`